net/xia/ppal_ether/main.c

```diff
@@ -497,6 +533,20 @@ static void free_neighs_by_interface(struct ether_interface *eint)
 	}
 }
 
+static void update_neighs_by_interface(struct ether_interface *eint)
+{
+	struct fib_xid_ether_main *mether;
+
+	ASSERT_RTNL();
+	rcu_read_lock();
+	list_for_each_entry_rcu(mether, &eint->list_interface_common_addr,
+				interface_common_addr) {
+		if (mether)
```

Comment:
The `if` test is not needed since `list_for_each_entry_rcu()` only calls its body for the items in the list.